### PR TITLE
chore(cli): pre-factor the sanitizer tests

### PR DESCRIPTION
**Pull request by mmastrac**
Moving tests around so that we can make #22413 smaller

### CODE REVIEW

File: tests/unit_node/process_test.ts [hunk 1]
```diff
@@ -226,6 +226,7 @@ Deno.test({
   },
 });
 
+// Only supported on Windows (but won't throw anywhere)
```
Comment (mmastrac) [inline tests/unit_node/process_test.ts:229]: Updated these two tests to correct expectations (and ensure that we do leak detection for this on all platforms)

File: tests/testdata/test/sanitizer/trace_ops_caught_error/main.ts [hunk 2]
```diff
@@ -1,3 +1,4 @@
+// https://github.com/denoland/deno/pull/16970
```
Comment (mmastrac) [inline tests/testdata/test/sanitizer/trace_ops_caught_error/main.ts:1]: Adding context

File: tests/testdata/test/sanitizer/ops_sanitizer_unstable.ts [hunk 3]
```diff
@@ -2,6 +2,7 @@ Deno.test("no-op", function () {});
 Deno.test({
   name: "leak interval",
   // regression test for sanitizer errors being swallowed with permissions.
+  // https://github.com/denoland/deno/pull/18550
```
Comment (mmastrac) [inline tests/testdata/test/sanitizer/ops_sanitizer_unstable.ts:5]: Adding context

File: tests/testdata/test/sanitizer/trace_ops_caught_error/main.ts (hunk 2 above)
Comment (dsherret) [reply tests/testdata/test/sanitizer/trace_ops_caught_error/main.ts:1]: It's a good idea to do this. Thanks!

**Review by dsherret — APPROVED** [review]
LGTM

**Pull request merged**